### PR TITLE
Activity Goal functionality and add command

### LINKED ISSUE #111: Implement Activity Goal Tracking

### LINKED ISSUE #112: Implement set-activity-goal command

### PR DESCRIPTION
**Pull request by AlWo223**
closes #112 , closes #111

### CODE REVIEW

File: src/main/java/athleticli/data/Goal.java [hunk 1]
```diff
@@ -52,7 +52,7 @@ public Timespan getTimespan() {
      * @param date  The date to be matched.
      * @return      Whether the date is between the timespan.
      */
-    public boolean checkDate(LocalDate date) {
+    public static boolean checkDate(LocalDate date, Timespan timespan) {
```
Comment (AlWo223) [inline src/main/java/athleticli/data/Goal.java:55]: I made this one static @skylee03 as this was easier to adapt to my existing code. Please let me know if it should not be static. Then I will adjust my code accordingly

Comment (skylee03) [reply src/main/java/athleticli/data/Goal.java:55]: > I made this one static @skylee03 as this was easier to adapt to my existing code. Please let me know if it should not be static. Then I will adjust my code accordingly

I think it's OK, but the JavaDoc also needs to be modified.

File: src/main/java/athleticli/data/activity/ActivityGoal.java [hunk 2]
```diff
@@ -1,13 +1,113 @@
...
+    }
+
+    /**
+     * Returns the current value of the activity goal metric.
+     * @param data The data containing the activity list.
+     * @return The current value of the activity goal metric.
+     */
+    public int getCurrentValue(Data data) throws IllegalStateException {
+        ActivityList activities = data.getActivities();
+        Class<?> activityClass = getActivityClass();
```
Comment (skylee03) [inline src/main/java/athleticli/data/activity/ActivityGoal.java:53]: Great use of `Class<?>`!

File: src/test/java/athleticli/data/activity/ActivityGoalTest.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        activityList.add(activityOutsidePeriod);
+
+        boolean expected = false;
+        boolean actual = activityGoal.isAchieved(data);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void setTargetValue() {
+    }
```
Comment (skylee03) [inline src/test/java/athleticli/data/activity/ActivityGoalTest.java:98]:
```suggestion
```

File: src/test/java/athleticli/data/activity/ActivityListTest.java [hunk 4]
```diff
@@ -16,17 +17,15 @@ class ActivityListTest {
...
     private Activity activitySecond;
 
 
     @BeforeEach
     void setUp() {
         activityList = new ActivityList();
-        dateSecond = LocalDateTime.of(2023, 10, 10, 23, 21);
-        dateFirst = LocalDateTime.of(2023, 10, 9, 23, 21);
+        LocalDateTime dateSecond = LocalDateTime.now();
+        LocalDateTime dateFirst = LocalDateTime.now().minusDays(1);
```
Comment (skylee03) [inline src/test/java/athleticli/data/activity/ActivityListTest.java:28]: Great use of `LocalDateTime.now()`!

File: src/main/java/athleticli/data/Goal.java [hunk 5]
```diff
@@ -52,7 +52,7 @@ public Timespan getTimespan() {
      * @param date  The date to be matched.
      * @return      Whether the date is between the timespan.
```
Comment (skylee03) [inline src/main/java/athleticli/data/Goal.java:53]:
```suggestion
     * @param date     The date to be matched.
     * @param timespan The timespan of the goal.
     * @return         Whether the date is between the timespan.
```

**Review by skylee03 — APPROVED** [review]
LGTM!

**Comment by AlWo223** [comment]
implements activity part of #73

**Pull request merged**